modules/rostests/apitests/gdi32/StretchBlt.c

```diff
@@ -18,91 +18,807 @@
+        expected[3] = 0x0000FF00;
+    }
+
+    test_StretchBlt_stretch(hdcDst, hdcSrc, &biDst, dstBuffer, srcBuffer,
+                             1, 0, -2, 2, 1, 1, -2, -2, expected, __LINE__, SrcTopDown, DstTopDown); // Case 0100 (4) - Vertical Flip.
+
+    if ((SrcTopDown || DstTopDown) && !(SrcTopDown && DstTopDown))
+    {
+        expected[0] = 0x0000FF00;
+        expected[1] = 0x000000FF;
+        expected[2] = 0xFF000000;
+        expected[3] = 0x00FF0000;
+    }
+    else
+    {
+        expected[0] = 0xFF000000;
+        expected[1] = 0x00FF0000;
+        expected[2] = 0x0000FF00;
+        expected[3] = 0x000000FF;
+    }
+
+    test_StretchBlt_stretch(hdcDst, hdcSrc, &biDst, dstBuffer, srcBuffer,
+                             1, 1, -2, -2, 0, 0, 2, 2, expected, __LINE__, SrcTopDown, DstTopDown); // Case 0011 (3) - Both Flip.
+
+    if ((SrcTopDown || DstTopDown) && !(SrcTopDown && DstTopDown))
+    {
+        expected[0] = 0xFF000000;
+        expected[1] = 0x00FF0000;
+        expected[2] = 0x0000FF00;
+        expected[3] = 0x000000FF;
+    }
+    else
+    {
+        expected[0] = 0x0000FF00;
+        expected[1] = 0x000000FF;
+        expected[2] = 0xFF000000;
+        expected[3] = 0x00FF0000;
+    }
+
+    test_StretchBlt_stretch(hdcDst, hdcSrc, &biDst, dstBuffer, srcBuffer,
+                             1, 1, -2, -2, 0, 1, 2, -2, expected, __LINE__, SrcTopDown, DstTopDown); // Case 0010 (2) - Horizontal Flip. 
+
+    if ((SrcTopDown || DstTopDown) && !(SrcTopDown && DstTopDown))
+    {
+        expected[0] = 0x000000FF;
+        expected[1] = 0x0000FF00;
+        expected[2] = 0x00FF0000;
+        expected[3] = 0xFF000000;
+    }
+    else
+    {
+        expected[0] = 0x00FF0000;
+        expected[1] = 0xFF000000;
+        expected[2] = 0x000000FF;
+        expected[3] = 0x0000FF00;
+    }
+
+    test_StretchBlt_stretch(hdcDst, hdcSrc, &biDst, dstBuffer, srcBuffer,
+                             1, 1, -2, -2, 1, 0, -2, 2, expected, __LINE__, SrcTopDown, DstTopDown); // Case 0001 (1) - Vertical Flip. 
+
+    if ((SrcTopDown || DstTopDown) && !(SrcTopDown && DstTopDown))
+    {
+        expected[0] = 0x00FF0000;
+        expected[1] = 0xFF000000;
+        expected[2] = 0x000000FF;
+        expected[3] = 0x0000FF00;
+
+    }
+    else
+    {
+        expected[0] = 0x000000FF;
+        expected[1] = 0x0000FF00;
+        expected[2] = 0x00FF0000;
+        expected[3] = 0xFF000000;
+    }
+
+    test_StretchBlt_stretch(hdcDst, hdcSrc, &biDst, dstBuffer, srcBuffer,
+                             2, 2, -2, -2, 2, 2, -2, -2, expected, __LINE__, SrcTopDown, DstTopDown); // Case 0000 (0) - No Flip. Special Case.
+
+
+    SelectObject(hdcSrc, oldSrc);
+    DeleteObject(bmpSrc);
+    DeleteDC(hdcSrc);
+
+    SelectObject(hdcDst, oldDst);
+    DeleteObject(bmpDst);
+    DeleteDC(hdcDst);
+
+    DeleteDC(hdcScreen);
 }
 
 
 START_TEST(StretchBlt)
 {
+    DPRINT1("\nStart of Generalized StretchBlt tests.\n");
```

Comment:
In apitests there is the `trace` macro.

Reply:
@learn-more, Thanks for pointing this out. I am not used to thinking in all lower case for macro's. I will update soon.

Reply:
Now changed to "trace" statements. Thanks.